File: AzDG.py
#! /usr/bin/python
# -*- coding: utf-8 -*-
# python 3.2

import hashlib
import base64
import time
import random

class AzDG:
	"""docstring for AzDG"""
	 __all__ = ["Encrypt", "Decrypt"]
    cipher = 'Private key'
	charset = 'utf-8'
	
	def __init__(self, cipher = None):
		if cipher != None:
			self.cipher = cipher

	def cipherEncode(self, sourceText):	
		cipherHash = hashlib.md5(self.cipher.encode(self.charset)).hexdigest().encode(self.charset)		
		encodeText = bytearray()
		for i in range(len(sourceText)):			
			encodeText.append(sourceText[i] ^ cipherHash[i%32])	
		return encodeText.decode('ISO-8859-1')
				
	def Encrypt(self, sourceText, charset = 'utf-8'):
		if charset != self.charset:
			self.charset = charset
		sourceText = sourceText.encode(self.charset)		
		noise = hashlib.md5(str(time.time()).encode(self.charset)).hexdigest().encode(self.charset)		
		encodeText = bytearray()
		for i in range(len(sourceText)):			
			encodeText.append(noise[i%32])	
			encodeText.append(sourceText[i] ^ noise[i%32])		
		return base64.b64encode(self.cipherEncode(encodeText).encode('ISO-8859-1')).decode(self.charset)
	
	def Decrypt(self, sourceText, charset = 'utf-8'):
		if charset != self.charset:
			self.charset = charset
		decodeSourceText = self.cipherEncode(base64.b64decode(sourceText.encode(self.charset)))		
		textLength = len(decodeSourceText)
		decodeText = bytearray()
		i = 0			
		while i < textLength:	
			decodeText.append(ord(decodeSourceText[i]) ^ ord(decodeSourceText[i+1]))	
			i += 2
		return decodeText.decode(self.charset)    

File: azdg.rb
#!/usr/bin/env ruby
# encoding: UTF-8
require 'digest'
require 'base64'

class AzDG
    attr_accessor :cipher
    def initialize(cipher = "2407Tw0kooco0wT7042")
        @cipherHash = Digest::MD5.hexdigest(cipher)
    end

    def cipherEncode(inputData)   
        outData = Array.new(inputData.length)
        for i in (0..inputData.length-1) 
            outData[i] = (inputData[i] ^ @cipherHash[i%32].ord)
        end    
        return outData
    end

    def Encrypt(inputData)  
        noise = Digest::MD5.hexdigest(Time.new.strftime("%Y-%m-%d %H:%M:%S"))
        aryInputData = inputData.bytes.to_a        
        outData = Array.new(aryInputData.length * 2)
        for i in (0..aryInputData.length-1) 
            outData[(i*2)] = noise[i%32].ord     
            outData[(i*2)+1] = aryInputData[i] ^ noise[i%32].ord 
        end        
        return Base64.encode64(cipherEncode(outData).pack('c*'))
    end

    def Decrypt(inputData) 
        decodeSourceText = cipherEncode(Base64.decode64(inputData).bytes.to_a)
        outData = Array.new()
        i = 0           
        while i < decodeSourceText.length do 
            outData.push(decodeSourceText[i].ord ^ decodeSourceText[i+1].ord)    
            i += 2
        end
        return outData.pack('c*').force_encoding('UTF-8')
    end
end     